Doc/using/cmdline.rst

```diff
@@ -411,11 +411,14 @@ Miscellaneous options
      nested imports).  Note that its output may be broken in multi-threaded
      application.  Typical usage is ``python3 -X importtime -c 'import
      asyncio'``.  See also :envvar:`PYTHONPROFILEIMPORTTIME`.
-   * ``-X dev`` enables the "developer mode": enable debug checks at runtime.
-     In short, ``python3 -X dev ...`` behaves as ``PYTHONMALLOC=debug PYTHONASYNCIODEBUG=1 python3
-     -W default -X faulthandler ...``, except that the :envvar:`PYTHONMALLOC`
-     and :envvar:`PYTHONASYNCIODEBUG` environment variables are not set in
-     practice. Developer mode:
+   * ``-X dev``: enable the "developer mode", enable runtime debug checks which
+     are too expensive to be enabled by default. It should not be more verbose
+     than the default if the code is correct: new warnings are only emitted
+     when an isuse is detected. In short, ``python3 -X dev ...`` behaves as
+     ``PYTHONMALLOC=debug PYTHONASYNCIODEBUG=1 python3 -W default -X faulthandler ...``,
+     except that the :envvar:`PYTHONMALLOC` and :envvar:`PYTHONASYNCIODEBUG`
+     environment variables are not set in practice. Effect of the developer
```

Comment:
Speaking of which, perhaps it would be nice to set those environment variables so that child processes launched using `subprocess` inherit them?

Reply:
Or perhaps let people set them themselves if they want to...

Reply:
It's a deliberate choice to not inherit the "developer mode" to child processes. If someone would like an inherited mode, I would prefer to add a new PYTHONDEVMODE=1 environment variable, since it's more natural to inherit environment modes.

It's the same for -W ignore vs PYTHONWARNINGS=ignore.

Reply:
Do you find that PYTHONDEVMODE=1 is needed?

Reply:
Perhaps `PYTHONXOPTIONS=dev`? That way it would more directly parallel the `-W` vs `PYTHONWARNINGS` arrangement.